Added ternary operator and unary plus operator

ggetz opened the pull request:
Added ternary conditional and unary plus operators, and also updated the tests.

pjcozzi commented on Specs/Scene/ExpressionSpec.js:
```diff
@@ -524,4 +534,12 @@ defineSuite([
         expect(expression.evaluate(undefined)).toEqual(false);
     });
+
+    it('evaluates ternary conditional', function() {
```
Add tests with more complex expressions, e.g., `(!(1 + 2 > 3)) ? (2 > 1 : 1 + 1 : 0) : (2 > 1 : -1 + -1 : 0)` should be `2`.

pjcozzi commented on Source/Scene/Expression.js:
```diff
@@ -261,7 +267,9 @@ define([
 
     function setEvaluateFunction(node) {
-        if (node._type === ExpressionNodeType.BINARY) {
+        if (node._type === ExpressionNodeType.TERNARY) {
```
Looking forward, from the AST perspective (not the tokenized perspective), perhaps `ExpressionNodeType.TERNARY` is better named `ExpressionNodeType.CONDITIONAL`.

ggetz commented:
OK, updated the tests. I also realized that some of the error checking with greater than/less than operations do not mimic JS behavior, so I removed it and updated the related specs.

Also changed `TERNARY` to `CONDITIONAL`

pjcozzi commented:
Did you push the commit?  I don't see it here.

ggetz commented:
Sorry, updated.

pjcozzi commented:
:+1:
